sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubMessageSerializer.java

```diff
@@ -107,14 +118,85 @@ public int getSize(Message amqpMessage) {
 
         final EventData eventData = (EventData) object;
         final Message message = Proton.message();
+        final byte[] body = eventData.getBody();
+
+        //TODO (conniey): support AMQP sequence and AMQP value.
+        message.setBody(new Data(new Binary(body)));
 
         if (eventData.getProperties() != null && !eventData.getProperties().isEmpty()) {
             message.setApplicationProperties(new ApplicationProperties(eventData.getProperties()));
         }
 
-        setSystemProperties(eventData, message);
+        if (message.getProperties() == null) {
+            message.setProperties(new Properties());
+        }
+
+        message.setMessageId(eventData.getMessageId());
+        message.setContentType(eventData.getContentType());
+        message.setCorrelationId(eventData.getCorrelationId());
+        message.setSubject(eventData.getRawAmqpMessage().getProperties().getSubject());
```

Comment:
nit:  We seem to be calling into `eventData.getRawAmqpMessage().getProperties()` quite a few times.   Should we consider using an intermediate variable to avoid the overhead of all the chained method calls?